### ISSUE TITLE
feat: add a visibility toggle of file name/ link

### ISSUE THREAD

**Issue by bguiz** [GitbookIO/integrations#179]
## What

- Add an additional parameter to the github-unfurl integration to specify whether or not to display the file name & link
  - This is optional, and if not specified, will default to true

## Why

Satisfy the following user stories:

- As a technical documenter, I would like to migrate existing code examples from inline (within markdown code blocks) to reference the same code within a github repo, and have it look the same as before.
  - In other words: If I didn't link to a file on github before, I would prefer to keep it the same

## Refs

- https://github.com/GitbookIO/integrations/pull/151

**Comment by addisonschultz** [comment]
Hey @bguiz, I've made some updates that should be now published and available ツ

Pull request: https://github.com/GitbookIO/integrations/pull/194

I've tested them both quite a bit, but please let me know if there's anything else I missed!

**Comment by bguiz** [comment]
Here's my experience using it, I'm trying to edit an existing page, specifically this bit:
https://github.com/hashgraph/hedera-docs/blob/9a6ab303a5/tutorials/more-tutorials/json-rpc-connections/hedera-json-rpc-relay.md?plain=1#L66-L78

(1) When I start a new Gitbook change request and edit an existing page it renders incorrectly in the editor. I'm expecting 5 lines from a `.env` file. Instead I just see 1 line, with a single character, `H`.

![image](https://github.com/GitbookIO/integrations/assets/1773785/fa491d21-c871-427c-be78-23332bdf8b38)

(2) The "show title & link" and "hide title & link" options work, but not sure why this isn't a toggle.
The "full width"/ "default width" toggle does not appear to do anything.

![image](https://github.com/GitbookIO/integrations/assets/1773785/8ce639a3-fc69-4156-9d21-f5c0c7f4dccd)

(3) The "configure github files" option appears to simply crash the editor. I cannot recover from this, pressing either button on the dialog appears to do nothing. It simply seems to stay on this URL `https://app.gitbook.com/o/-LuC734MpqlgwA6zyhAO/s/-LsUmP_7NoX2exvbv_Ps/~/integrations/github-files` and refresh itself. I need to hit "bacK' to go back to the URL of my change request, and then hit "refresh".

![image](https://github.com/GitbookIO/integrations/assets/1773785/f5a27094-4961-4e33-8415-bcfa78f64109)

**Comment by bguiz** [comment]
@addisonschultz I suspect that one possible cause for the error described in (3) above is that the editor sees an old/ previous syntax for this plugin, and perhaps the fix might be a manual git commit to the new/ updated syntax for this plugin.

```
{% @github-files/github-code-block url="https://github.com/hashgraph/hedera-json-rpc-relay/blob/f9d5ebaa80/docs/examples/.env.mainnet.sample#L1-L5" %}
```

❓ if this is the case, could you pls provide the new/ updated syntax? I'd like to try it out.

**Comment by bguiz** [comment]
@addisonschultz Also, more broadly, I think it might be good if there was a `README.md` file per plugin, e.g. one here:
https://github.com/GitbookIO/integrations/tree/main/integrations/github

I'm thinking of something a bit more developer-oriented (rather than user-oriented) than the summary section of the `gitbook-manifest.yaml` file, e.g.:
https://github.com/GitbookIO/integrations/blob/main/integrations/github/gitbook-manifest.yaml

... and specifically that these READMEs includes some example syntax of what the plugin might output.

**Comment by addisonschultz** [comment]
Hey @bguiz 

Let me look into this a bit more. For some context on the points above:

1. Looks like an issue with the way it's fetching the data. To help me debug a bit more, was that file ever just a single character "H"?
2. We unfortunately don't have a toggle for our `options` in `block` controls yet - but definitely something we'd want to add in the future. The reason the full-width isn't working in this context, is because it looks like it's inside a tab component
3. Also looks like a pretty weird bug - I'll try to reproduce this (please let me know if this still happens). I don't believe the syntax is causing this (But will also look into it) 

Readme's (and videos on the integrations) are in our backlog/radar!

**Comment by bguiz** [comment]
> 1. Looks like an issue with the way it's fetching the data. To help me debug a bit more, was that file ever just a single character "H"?

Nope, the target file has always been a `.env` file similar to its current form (5 key value pairs on 5 lines)

> 2. We unfortunately don't have a toggle for our `options` in `block` controls yet - but definitely something we'd want to add in the future. The reason the full-width isn't working in this context, is because it looks like it's inside a tab component

Ah I see, got it!

> 3. Also looks like a pretty weird bug - I'll try to reproduce this (please let me know if this still happens). I don't believe the syntax is causing this (But will also look into it)

OK, just tried again and it is behaving in the exact same way. 

> Readme's (and videos on the integrations) are in our backlog/radar!

Awesome, thanks - looking forward to them!

**Comment by addisonschultz** [comment]
Hey @bguiz - Are you able to remove the line references in your files for the ones that aren't working properly? I see you're referencing lines 1-5 (the entire file), but also notice this is causing it to not pick everything up for some reason. I haven't figured out why yet, but it seems to work when the lines are removed

**Comment by bguiz** [comment]
I guess this might be a solution for (1), but currently not possible, because of issue (3).

Not possible in the gitbook app that is, I could do that via a direct git commit to the repo, but that feels more like a workaround than a solution.

Also, in this particular page and reference file from github, removing the line numbers would work
... however in other pages I explicitly want to be able to reference a specific range of line numbers.

> Hey @bguiz - Are you able to remove the line references in your files for the ones that aren't working properly? I see you're referencing lines 1-5 (the entire file), but also notice this is causing it to not pick everything up for some reason.
